test/e2e/serial/workload_placement_tmpol_test.go

```diff
@@ -0,0 +1,375 @@
+			pod.Spec.SchedulerName = schedulerName
+			pod.Spec.NodeSelector = map[string]string{
+				multiNUMALabel: "2",
+			}
+			pod.Spec.Containers = append(pod.Spec.Containers, pod.Spec.Containers[0])
+			pod.Spec.Containers[0].Name = "testcnt-0"
+			pod.Spec.Containers[0].Resources.Limits = podRes[0]
+			pod.Spec.Containers[1].Name = "testcnt-1"
+			pod.Spec.Containers[1].Resources.Limits = podRes[1]
+
+			requiredRes := e2ereslist.FromGuaranteedPod(*pod)
+
+			numaZonesRequired := 2
+
+			By(fmt.Sprintf("filtering available nodes with at least %d NUMA zones", numaZonesRequired))
+			nrtCandidates := e2enrt.FilterZoneCountEqual(nrts, numaZonesRequired)
+			if len(nrtCandidates) < hostsRequired {
+				Skip(fmt.Sprintf("not enough nodes with %d NUMA Zones: found %d", numaZonesRequired, len(nrtCandidates)))
+			}
+			By("filtering available nodes with allocatable resources on each NUMA zone that can match request")
+			nrtCandidates = e2enrt.FilterAnyZoneMatchingResources(nrtCandidates, requiredRes)
+			if len(nrtCandidates) < hostsRequired {
+				Skip(fmt.Sprintf("not enough nodes with NUMA zones each of them can match requests: found %d", len(nrtCandidates)))
+			}
+
+			candidateNodeNames := e2enrt.AccumulateNames(nrtCandidates)
+			// nodes we have now are all equal for our purposes. Pick one at random
+			// TODO: make sure we can control this randomness using ginkgo seed or any other way
+			targetNodeName, ok := candidateNodeNames.PopAny()
+			Expect(ok).To(BeTrue(), "cannot select a target node among %#v", candidateNodeNames.List())
+			unsuitableNodeNames := candidateNodeNames.List()
+
+			By(fmt.Sprintf("selecting target node %q and unsuitable nodes %#v (random pick)", targetNodeName, unsuitableNodeNames))
+
+			padInfo := paddingInfo{
+				pod:                 pod,
+				targetNodeName:      targetNodeName,
+				unsuitableNodeNames: unsuitableNodeNames,
+				unsuitableFreeRes:   unsuitableFreeRes,
+			}
+
+			By("Padding nodes to create the test workload scenario")
+			paddingPods := setupPadding(fxt, nrtList, padInfo)
+
+			By("waiting for ALL padding pods to go running - or fail")
+			failedPods := e2ewait.ForPodListAllRunning(fxt.Client, paddingPods)
+			for _, failedPod := range failedPods {
+				// ignore errors intentionally
+				_ = objects.LogEventsForPod(fxt.K8sClient, failedPod.Namespace, failedPod.Name)
+			}
+			Expect(failedPods).To(BeEmpty())
+
+			// TODO: smarter cooldown
+			time.Sleep(18 * time.Second)
+			for _, unsuitableNodeName := range unsuitableNodeNames {
+				dumpNRTForNode(fxt.Client, unsuitableNodeName, "unsuitable")
+			}
+			dumpNRTForNode(fxt.Client, targetNodeName, "target")
+
+			By("checking the resource allocation as the test starts")
+			nrtListInitial, err := e2enrt.GetUpdated(fxt.Client, nrtList, 1*time.Minute)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("running the test pod")
+			if data, err := yaml.Marshal(pod); err != nil {
+				klog.Infof("Pod:\n%s", data)
```

Comment:
I don't see we use data in any other place, is this needed for debugging?

Reply:
yep